spec/controllers/projects_controller_spec.rb

```diff
@@ -1,5 +1,37 @@
 require 'rails_helper'
 
 RSpec.describe ProjectsController, type: :controller do
+  let(:project) { create :project }
 
+  before { sign_in :user, project.user }
+
+  shared_examples 'authentication' do
+    context 'not singed in' do
+      before { sign_out :user }
+
+      it 'can not access the action' do
+        is_expected.not_to have_http_status :ok
+      end
+    end
+  end
+
+  describe '.index' do
+    subject { get :index }
+
+    it 'renders the project index page' do
+      is_expected.to have_http_status :ok
+    end
+
+    include_examples 'authentication'
+  end
+
+  describe '.show' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.